### PR TITLE
special functions for missing

### PR DESCRIPTION
**Pull request by fredrikekre**
See https://github.com/JuliaData/Missings.jl/pull/85

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -624,6 +624,10 @@ end
         @test lgamma(-Inf*im) === -Inf - Inf*im
         @test lgamma(Inf + Inf*im) === lgamma(NaN + 0im) === lgamma(NaN*im) === NaN + NaN*im
     end
+
+    @testset "missing" begin
+        @test gamma(missing) === missing
+        @test lgamma(missing) === missing
```
Comment (Keno) [inline test/runtests.jl]: missing `end`?

**Comment by Keno** [comment]
Do we want to provide these methods also for `[l]beta` and `factorial`?

**Comment by ararslan** [comment]
Yes, definitely for beta and all of the other ones in that family. We may also want to consider adding methods for the Bessel functions.

**Comment by fredrikekre** [comment]
`factorial` is Base-owned so that should probably be defined in Base. I can add the other ones for SpecialFunctions though.

File: test/runtests.jl [hunk 2]
```diff
@@ -641,3 +641,10 @@ end
 
     @test beta(big(1.0),big(1.2)) ≈ beta(1.0,1.2) rtol=4*eps()
 end
+
+@test "missing data" begin
```
Comment (Keno) [inline test/runtests.jl]: `@testset`

**Comment by fredrikekre** [comment]
Can we put julia REQUIRE to alpha? Otherwise we have to include Missings as a dependency.

File: test/runtests.jl (hunk 2 above)
Comment (fredrikekre) [reply test/runtests.jl]: Sigh... to late 💤

**Comment by Keno** [comment]
Yes, we can do that. In that case, we can also drop Compat at the same time.

**Comment by ararslan** [comment]
Alternatively, just wrap the definitions and tests in
```julia
if isdefined(Base, :missing)
    # ...
end
```

**Comment by ararslan** [comment]
Let's add methods for:

* `beta`
* `digamma`
* `erf`
* `erfc`
* `erfcinv`
* `erfcx`
* `erfi`
* `erfinv`
* `eta`
* `gamma`
* `invdigamma`
* `lbeta`
* `lfact`
* `lfactorial`
* `lgamma`
* `polygamma`
* `trigamma`

Does that seem like a reasonable selection? It basically just omits Bessel-family functions.

File: src/SpecialFunctions.jl [hunk 3]
```diff
@@ -76,4 +76,11 @@ include("sincosint.jl")
 include("gamma.jl")
 include("deprecated.jl")
 
+@static if isdefined(Base, :missing)
+    for f in (:beta, :digamma, :erf, :erfc, :erfcinv, :erfcx, :erfi, :erfinv, :eta, :gamma,
```
Comment (ararslan) [inline src/SpecialFunctions.jl]: We'll have to structure things a little differently for some of these. In particular, `beta` takes two arguments, and we'd want the result to be missing if either or both is missing, e.g.
```julia
beta(a::Missing, b::Number) = missing
beta(a::Number, b::Missing) = missing
beta(a::Missing, b::Missing) = missing
```
Further, `polygamma(m, x)` is the `m`th derivative of `lgamma(x)`, so we should instead define
```julia
polygamma(m::Integer, x::Missing) = missing
```
but don't let `m` be missing.

**Comment by Keno** [comment]
Looks good from my side. @ararslan?

**Comment by ararslan** [comment]
Seems okay to me but I've asked @simonbyrne for confirmation on what makes sense here.

**Comment by simonbyrne** [comment]
LGTM

**Pull request merged**